File: Sort_algo/QuickSort.cpp
#include<bits/stdc++.h>
using namespace std;
void printArray(int arr[],int n)
{
    for(int i=0;i<n;i++)
    {
        cout << arr[i] << " " ;
    }
    cout << "\n";
}
void swapping(int *xp,int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

int partition(int arr[],int low , int high)
{
    int pivot = arr[high];
    int i = low - 1;
    for(int j=low;j<=high-1;j++)
    {
        if(arr[j] < pivot)
        {
            i++;
            swapping(&arr[i],&arr[j]);
        }
    }
    swapping(&arr[i+1],&arr[high]);
    return (i+1);
}

void quickSort(int arr[],int low,int high)
{
    if(low < high)
    {
        int pi = partition(arr,low,high);

        quickSort(arr,low,pi-1);
        quickSort(arr,pi+1,high);
    }
}
int main()
{
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }

    printArray(arr,n);
    quickSort(arr,0,n-1);
    printArray(arr,n);
    return 0;
}

File: Sort_algo/quickSort.py
arr = []
def input_arr():
    n = int(input("Enter size of array:\n"))
    for i in range(0,n):
        ele = int(input("Enter Elements: \n"))
        arr.append(ele)

def quick_sort(arr,low,high):
    if low < high:
        pi = partition(arr,low,high)

        quick_sort(arr,low,pi - 1)
        quick_sort(arr,pi+1,high)

def partition(arr,low,high):
    pivot = arr[high]
    i = low - 1
    for j in range (low,high):
        if pivot > arr[j]:
            i+=1
            arr[i],arr[j] = arr[j],arr[i]
    arr[i+1],arr[high] = arr[high],arr[i+1]
    return (i+1)



input_arr()
n = len(arr)
print("Before\n")
print(arr)
quick_sort(arr,0,n-1)
print("after\n")
print(arr)